gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.Serializable;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class that represents Google Cloud DNS record set.
+ *
+ * <p> A unit of data that will be returned by the DNS servers.
+ *
+ * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets">Google Cloud DNS
+ * documentation</a>
+ */
+public class DnsRecord implements Serializable {
+
+  private static final long serialVersionUID = 2016011914302204L;
+  private final String name;
+  private final List<String> rrdatas;
+  private final Integer ttl;
+  private final DnsRecordType type;
+  private final String zoneName;
+  private final Long zoneId;
+
+  /**
+   * A private constructor. Obtain an instance using {@link DnsRecord#Builder}.
```

Comment:
Docs on private method won't be seen by users so I would avoid them unless it's noteworthy implementation details.